src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -17,9 +17,13 @@
  * under the License.
  */
 
-import { cloneDeep, get, omit } from 'lodash';
+import { cloneDeep, get, omit, has } from 'lodash';
 
-function migrateIndexPattern(doc) {
+const executeMigrations = (fns, doc) => {
```

Comment:
It seems like it would be more direct to just put your `migrateIndexPattern` directly into the migrations definitions the way `'7.0.0` was done:

```js
    '7.1.0': (doc) => {
      // ... your logic here...
    },
```

I wouldn't try to abstract this until it becomes necessary (e.g. if someone writes another visualization migration in 7.0.1, maybe they can introduce something like this? But I wouldn't introduce it for a single migration.